api/envoy/data/accesslog/v3/accesslog.proto

```diff
@@ -236,6 +256,9 @@ message AccessLogCommon {
   // For HTTP: Total number of bytes received from the upstream by the http stream.
   // For TCP: Total number of bytes sent to the upstream by the tcp proxy.
   uint64 upstream_wire_bytes_received = 32;
+
+  // The type of the access log, which indicates when the log was recorded.
```

Comment:
Please describe what's the default behavior.
One thing to note, at the moment there isn't a per-enum-value configuration, but if you think it will be added in the near future, please consider using a message type (with `oneof`) instead of an enum.

Reply:
Thanks @adisuissa, I have update the doc with more info about this behavior. Shortly, for HCM/TcpProxy/Router the value will be as the new ACCESS_LOG_TYPE substitution value. Otherwise - an empty string (for flows which are currently not supported by this substitution string, for example - listener access logs).

I'm not sure I understood the second part of your comment, the type of this field is an enum because it's unique for an access log record. Anyways, can you please elaborate more on what you meant?

Reply:
Thanks.
In the second part I meant that if one of the types will need an additional config, e.g., `UpstreamPeriodic` will need a duration, then it may make sense to convert this to a message, with `oneof` between the different enum values that are specified, and each will be a message.
If no value-specific config is needed, then enum should be kept.

Reply:
As far as I can tell, for telling the type of the access log itself, there's no other config data required here, please LMK if I have missed something